### ISSUE TITLE
v0.8(.1) segment violation panic

### ISSUE THREAD

**Issue by varod** [42wim/matterbridge#91]
### Which version of matterbridge are you using?
V0.8 and V0.8.1

### If you're having problems with mattermost please specify mattermost version. 
V3.5

#### Use logs from running ```matterbridge -debug``` if possible.
Just after few seconds after the start of matterbridge it crashes. I see bridge's irc user connected and logs seems to say everything is ok.
Same configuration was working well with v0.7 before mattermost upgrade.

```
running version 0.8.0
starting gateway "gateway1"
INFO[2016-11-23T20:37:38+01:00] Starting bridge: irc.freenode channel: #pepper&carrot 
INFO[2016-11-23T20:37:38+01:00] Starting bridge: mattermost.work channel: irc 
INFO[2016-11-23T20:37:38+01:00] Connecting irc.freenode.net:6697              module=irc
INFO[2016-11-23T20:37:46+01:00] Connection succeeded                          module=irc
INFO[2016-11-23T20:37:46+01:00] irc.freenode: joining #pepper&carrot         
INFO[2016-11-23T20:37:46+01:00] Connecting pepperbot (team: peppercarrot) on framateam.org  module=mattermost
INFO[2016-11-23T20:37:47+01:00] Connection succeeded                          module=mattermost
INFO[2016-11-23T20:37:47+01:00] mattermost.work: joining irc                 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x596ff4]

goroutine 57 [running]:
panic(0x8ede60, 0xc42000c0a0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/42wim/matterbridge/matterclient.(*MMClient).parseActionPost(0xc420404990, 0xc4203a0dc0)
	/go/src/github.com/42wim/matterbridge/matterclient/matterclient.go:265 +0x184
github.com/42wim/matterbridge/matterclient.(*MMClient).parseMessage(0xc420404990, 0xc4203a0dc0)
	/go/src/github.com/42wim/matterbridge/matterclient/matterclient.go:240 +0x87
github.com/42wim/matterbridge/matterclient.(*MMClient).WsReceiver(0xc420404990)
	/go/src/github.com/42wim/matterbridge/matterclient/matterclient.go:223 +0x639
created by github.com/42wim/matterbridge/bridge/mattermost.(*Bmattermost).Connect
	/go/src/github.com/42wim/matterbridge/bridge/mattermost/mattermost.go:77 +0x36e

```
any idea what goes wrong ? 
Thanks in advance for you support.

**Comment by varod** [comment]
```
running version 0.8.0
starting gateway "gateway1"
INFO[2016-11-23T20:37:38+01:00] Starting bridge: irc.freenode channel: #pepper&carrot 
INFO[2016-11-23T20:37:38+01:00] Starting bridge: mattermost.work channel: irc 
INFO[2016-11-23T20:37:38+01:00] Connecting irc.freenode.net:6697              module=irc
INFO[2016-11-23T20:37:46+01:00] Connection succeeded                          module=irc
INFO[2016-11-23T20:37:46+01:00] irc.freenode: joining #pepper&carrot         
INFO[2016-11-23T20:37:46+01:00] Connecting pepperbot (team: peppercarrot) on framateam.org  module=mattermost
INFO[2016-11-23T20:37:47+01:00] Connection succeeded                          module=mattermost
INFO[2016-11-23T20:37:47+01:00] mattermost.work: joining irc                 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x596ff4]

goroutine 57 [running]:
panic(0x8ede60, 0xc42000c0a0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/42wim/matterbridge/matterclient.(*MMClient).parseActionPost(0xc420404990, 0xc4203a0dc0)
	/go/src/github.com/42wim/matterbridge/matterclient/matterclient.go:265 +0x184
github.com/42wim/matterbridge/matterclient.(*MMClient).parseMessage(0xc420404990, 0xc4203a0dc0)
	/go/src/github.com/42wim/matterbridge/matterclient/matterclient.go:240 +0x87
github.com/42wim/matterbridge/matterclient.(*MMClient).WsReceiver(0xc420404990)
	/go/src/github.com/42wim/matterbridge/matterclient/matterclient.go:223 +0x639
created by github.com/42wim/matterbridge/bridge/mattermost.(*Bmattermost).Connect
	/go/src/github.com/42wim/matterbridge/bridge/mattermost/mattermost.go:77 +0x36e
root@vps187720:/opt/matterbridge# ./matterbridge-linux64 -debug
INFO[2016-11-23T20:59:06+01:00] enabling debug                               
running version 0.8.0
starting gateway "gateway1"
INFO[2016-11-23T20:59:06+01:00] Starting bridge: irc.freenode channel: #pepper&carrot 
INFO[2016-11-23T20:59:06+01:00] Starting bridge: mattermost.work channel: irc 
INFO[2016-11-23T20:59:06+01:00] Connecting irc.freenode.net:6697              module=irc
2016/11/23 20:59:06 --> NICK |||
2016/11/23 20:59:06 --> USER ||| 0.0.0.0 0.0.0.0 :|||
2016/11/23 20:59:06 <-- :rajaniemi.freenode.net NOTICE * :*** Looking up your hostname...
2016/11/23 20:59:06 <-- :rajaniemi.freenode.net NOTICE * :*** Checking Ident
2016/11/23 20:59:06 <-- :rajaniemi.freenode.net NOTICE * :*** Found your hostname
2016/11/23 20:59:16 <-- :rajaniemi.freenode.net NOTICE * :*** No Ident response
2016/11/23 20:59:16 <-- :rajaniemi.freenode.net 001 ||| :Welcome to the freenode Internet Relay Chat Network |||
DEBU[2016-11-23T20:59:16+01:00] Registering callbacks                         module=irc
DEBU[2016-11-23T20:59:16+01:00] ":rajaniemi.freenode.net 001 ||| :Welcome to the freenode Internet Relay Chat Network |||"  module=irc
2016/11/23 20:59:16 <-- :rajaniemi.freenode.net 002 ||| :Your host is rajaniemi.freenode.net[195.148.124.79/6697], running version ircd-seven-1.1.4
2016/11/23 20:59:16 <-- :rajaniemi.freenode.net 003 ||| :This server was created Thu Sep 22 2016 at 20:51:02 UTC
2016/11/23 20:59:16 <-- :rajaniemi.freenode.net 004 ||| rajaniemi.freenode.net ircd-seven-1.1.4 DOQRSZaghilopswz CFILMPQSbcefgijklmnopqrstvz bkloveqjfI
2016/11/23 20:59:16 <-- :rajaniemi.freenode.net 005 ||| CHANTYPES=# EXCEPTS INVEX CHANMODES=eIbq,k,flj,CFLMPQScgimnprstz CHANLIMIT=#:120 PREFIX=(ov)@+ MAXLIST=bqeI:100 MODES=4 NETWORK=freenode KNOCK STATUSMSG=@+ CALLERID=g :are supported by this server
2016/11/23 20:59:16 <-- :rajaniemi.freenode.net 005 ||| CASEMAPPING=rfc1459 CHARSET=ascii NICKLEN=16 CHANNELLEN=50 TOPICLEN=390 ETRACE CPRIVMSG CNOTICE DEAF=D MONITOR=100 FNC TARGMAX=NAMES:1,LIST:1,KICK:1,WHOIS:1,PRIVMSG:4,NOTICE:4,ACCEPT:,MONITOR: :are supported by this server
INFO[2016-11-23T20:59:16+01:00] Connection succeeded                          module=irc
INFO[2016-11-23T20:59:16+01:00] irc.freenode: joining #pepper&carrot         
INFO[2016-11-23T20:59:16+01:00] Connecting pepperbot (team: peppercarrot) on framateam.org  module=mattermost
DEBU[2016-11-23T20:59:16+01:00] trying login peppercarrot pepperbot framateam.org  module=matterclient
DEBU[2016-11-23T20:59:16+01:00] ":||| MODE ||| :+Zi"                          module=irc
DEBU[2016-11-23T20:59:17+01:00] ":NickServ!NickServ@services. NOTICE ||| :This nickname is registered. Please choose a different nickname, or identify via \x02/msg NickServ identify <password>\x02."  module=irc
DEBU[2016-11-23T20:59:18+01:00] initUser(): found our team peppercarrot (id: gpfzgmz3qjrx8c8cbxn13eieqh)  module=matterclient
DEBU[2016-11-23T20:59:18+01:00] WsClient: making connection                   module=matterclient
INFO[2016-11-23T20:59:18+01:00] Connection succeeded                          module=mattermost
INFO[2016-11-23T20:59:18+01:00] mattermost.work: joining irc                 
DEBU[2016-11-23T20:59:18+01:00] Not joining kqqmrrm9hifntehbmdpdf4nnic already joined.  module=matterclient
DEBU[2016-11-23T20:59:18+01:00] Choosing API based Mattermost connection: true  module=mattermost
DEBU[2016-11-23T20:59:18+01:00] WsReceiver: model.WebSocketEvent{Event:"hello", Data:map[string]interface {}{"server_version":"3.5.0.3.5.0.aeff72451d9afd37f87d9a7d99a4fd10"}, Broadcast:(*model.WebsocketBroadcast)(0xc4203fd480), PreComputeJson:[]uint8(nil)}  module=matterclient
DEBU[2016-11-23T20:59:18+01:00] WsReceiver: model.WebSocketEvent{Event:"status_change", Data:map[string]interface {}{"status":"online", "user_id":"xzs47of4ufft7cyibzfnw19hgr"}, Broadcast:(*model.WebsocketBroadcast)(0xc4203fd600), PreComputeJson:[]uint8(nil)}  module=matterclient
DEBU[2016-11-23T20:59:22+01:00] ":|||!~|||@57.ip-149-202-44.eu JOIN #pepper&carrot"  module=irc
DEBU[2016-11-23T20:59:22+01:00] ":rajaniemi.freenode.net 332 ||| #pepper&carrot :http://www.peppercarrot.com , the open webcomic  = ^ . ^ = , room linked to Telegram group: https://telegram.me/joinchat/BLVsYz_DIz-S-TJZB9XW7A"  module=irc
DEBU[2016-11-23T20:59:22+01:00] 333: Topic set by deevad [~Deevad@cou82-2-88-180-132-198.fbx.proxad.net] [2016-07-30 17:22:55 +0200 CEST]  module=irc
DEBU[2016-11-23T20:59:22+01:00] ":rajaniemi.freenode.net 333 ||| #pepper&carrot deevad!~Deevad@cou82-2-88-180-132-198.fbx.proxad.net 1469892175"  module=irc
DEBU[2016-11-23T20:59:22+01:00] ":rajaniemi.freenode.net 353 ||| = #pepper&carrot :||| Jookia TheThing_ Szkodnix raspbeguy francoist irina deevad raghukamath ^^^^ lvgx M1dgard CalimeroTeknik VickySteeves[m] michael[m] Remramm arossdotme Nuparu__ mensi1 Dezponia gleki cmaloney chll valvin davidb2111 sn0wmonster bloodywing chaiomanot Mylene sir_anok1"  module=irc
DEBU[2016-11-23T20:59:22+01:00] ":rajaniemi.freenode.net 366 ||| #pepper&carrot :End of /NAMES list."  module=irc
DEBU[2016-11-23T20:59:42+01:00] handlePrivMsg() valvin :) &irc.Event{Code:"PRIVMSG", Raw:":valvin!~valvin@45.32.239.240 PRIVMSG #pepper&carrot ::)", Nick:"valvin", Host:"45.32.239.240", Source:"valvin!~valvin@45.32.239.240", User:"~valvin", Arguments:[]string{"#pepper&carrot", ":)"}, Connection:(*irc.Connection)(0xc420156160)}  module=irc
DEBU[2016-11-23T20:59:42+01:00] Sending message from #pepper&carrot on irc.freenode to gateway  module=irc
DEBU[2016-11-23T20:59:42+01:00] ":valvin!~valvin@45.32.239.240 PRIVMSG #pepper&carrot ::)"  module=irc
DEBU[2016-11-23T20:59:42+01:00] Sending config.Message{Text:":)", Channel:"irc", Username:"valvin", Origin:"freenode", FullOrigin:"irc.freenode", Protocol:"irc", Avatar:""} from irc.freenode (#pepper&carrot) to mattermost.work (irc) 
DEBU[2016-11-23T20:59:42+01:00] Receiving config.Message{Text:":)", Channel:"irc", Username:"valvin", Origin:"freenode", FullOrigin:"irc.freenode", Protocol:"irc", Avatar:""}  module=mattermost
DEBU[2016-11-23T20:59:42+01:00] WsReceiver: model.WebSocketEvent{Event:"posted", Data:map[string]interface {}{"channel_display_name":"Komona (en)", "channel_type":"O", "post":"{\"id\":\"q8zan1xwdirzmdj45k7r8p4bqc\",\"create_at\":1479931182492,\"update_at\":1479931182492,\"delete_at\":0,\"user_id\":\"xzs47of4ufft7cyibzfnw19hgr\",\"channel_id\":\"kqqmrrm9hifntehbmdpdf4nnic\",\"root_id\":\"\",\"parent_id\":\"\",\"original_id\":\"\",\"message\":\"**valvin** : :)\",\"type\":\"\",\"props\":{},\"hashtags\":\"\",\"pending_post_id\":\"\"}", "sender_name":"pepperbot", "team_id":"gpfzgmz3qjrx8c8cbxn13eieqh"}, Broadcast:(*model.WebsocketBroadcast)(0xc420372c80), PreComputeJson:[]uint8(nil)}  module=matterclient
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x596ff4]

goroutine 57 [running]:
panic(0x8ede60, 0xc420012090)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/42wim/matterbridge/matterclient.(*MMClient).parseActionPost(0xc420404bd0, 0xc42039cf00)
	/go/src/github.com/42wim/matterbridge/matterclient/matterclient.go:265 +0x184
github.com/42wim/matterbridge/matterclient.(*MMClient).parseMessage(0xc420404bd0, 0xc42039cf00)
	/go/src/github.com/42wim/matterbridge/matterclient/matterclient.go:240 +0x87
github.com/42wim/matterbridge/matterclient.(*MMClient).WsReceiver(0xc420404bd0)
	/go/src/github.com/42wim/matterbridge/matterclient/matterclient.go:223 +0x639
created by github.com/42wim/matterbridge/bridge/mattermost.(*Bmattermost).Connect
	/go/src/github.com/42wim/matterbridge/bridge/mattermost/mattermost.go:77 +0x36e
```

**Comment by 42wim** [comment]
Fixed in v0.8 branch.
Do a 
```
git pull
git checkout v0.8
go build
```

**Comment by varod** [comment]
it works well thanks you @42wim for your support and your reactivity !!! 20minutes to solve my issue :)
